bin/reth/src/optimism.rs

```diff
@@ -14,7 +21,37 @@ fn main() {
         std::env::set_var("RUST_BACKTRACE", "1");
     }
 
-    if let Err(err) = reth::cli::run() {
+    if let Err(err) = Cli::<RollupArgs>::parse().run(|builder, rollup_args| async {
+        let NodeHandle { node, node_exit_future } = builder
+            .with_types(OptimismNode::default())
+            .with_components(OptimismNode::components(rollup_args.clone()))
+            .launch()
+            .await?;
+
+        // If `enable_genesis_walkback` is set to true, the rollup client will need to
+        // perform the derivation pipeline from genesis, validating the data dir.
+        // When set to false, set the finalized, safe, and unsafe head block hashes
+        // on the rollup client using a fork choice update. This prevents the rollup
+        // client from performing the derivation pipeline from genesis, and instead
+        // starts syncing from the current tip in the DB.
+        if node.chain_spec().is_optimism() && !rollup_args.enable_genesis_walkback {
+            let client = node.rpc_server_handles.auth.http_client();
+            if let Ok(Some(head)) = node.provider.latest_header() {
+                reth_rpc_api::EngineApiClient::<OptimismEngineTypes>::fork_choice_updated_v2(
+                    &client,
+                    reth_rpc_types::engine::ForkchoiceState {
+                        head_block_hash: head.hash(),
+                        safe_block_hash: head.hash(),
+                        finalized_block_hash: head.hash(),
+                    },
+                    None,
+                )
+                .await?;
+            }
+        }
+
+        node_exit_future.await
```

Comment:
Mental note we should explore integrating Magi or parts of it https://github.com/a16z/magi/blob/82e2c51fb7a62c48d505d222b11d1ff5f784d129/src/engine/api.rs#L210